### PR TITLE
Change object ids to uuid's

### PR DESCRIPTION
**Pull request by ApproximateIdentity**
This changes the object ids to uuids. It seems to work to me, but I might have missed something. I fixed some tests and I skipped others. The ones I skipped I think probably don't apply anymore (since they are basically testing the integer generation). Take a look and see if you agree.

I haven't yet converted the example pages, but I'll do that once you guys are good with this.

### CODE REVIEW

File: js/objects/tests/test-button-system-commands-with-preload.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ describe('Button System Command Tests', () => {
             assert.exists(foundEl);
             assert.exists(foundEl.model);
         });
-        it('Can get card view by its id', () => {
+        it.skip('Can get card view by its id', () => {
```
Comment (dkrasner) [inline js/objects/tests/test-button-system-commands-with-preload.js:32]: why would this test need to be skipped?

**Comment by dkrasner** [comment]
@ApproximateIdentity this all makes sense to me

there are three things left to do imo:

* rebase over `master`   I merged in the [current PR](https://github.com/dkrasner/Simpletalk/pull/33) which deals with ids as well and it would be good to make sure all the tests still pass and the thing works
* remove everything that is related to the choice to use old ids or make news one [here](https://github.com/dkrasner/Simpletalk/blob/thomas-uuid-id/js/objects/utils/serialization.js#L234)   if all ids are unique i don't see why we would ever need to make this choice moving fwd (usually done on importing a world)
* update/re-serialize the examples

File: js/objects/tests/test-button-system-commands-with-preload.js (hunk 1 above)
Comment (ApproximateIdentity) [reply js/objects/tests/test-button-system-commands-with-preload.js:32]: you're right it doesn't need to be skipped i just need to change the logic not to depend on the hard-coded 2 value which depended on the old setup. i can fix that up

Comment (ApproximateIdentity) [reply js/objects/tests/test-button-system-commands-with-preload.js]: Fixed here: https://github.com/dkrasner/Simpletalk/pull/37/commits/de96a2c0355648cf56d519818d213fc9b174f2bf

**Comment by darth-cheney** [comment]
@ApproximateIdentity Is there any way to get a shorter hash for the IDs like the Notion people were using in their article?

**Comment by ApproximateIdentity** [comment]
> @ApproximateIdentity Is there any way to get a shorter hash for the IDs like the Notion people were using in their article?

@darth-cheney Since these are uniq we could do a couple things.

1. One is to not have actual UUIDs, but to have shorter strings. The shorter the higher likelihood of collision so we'd have to think about that.
2. Another option is to allow the referencing of a hash by any shorter prefix as long as it is uniq (so that there aren't multiple full hashes that have the same prefix).

I kind of think the second way is probably better, but I think that it is kind of orthogonal to this pr. If we were to implement that I think we should do that separately and I could look into that next.

But one thing to consider is how often are we accessing hashes in code explicitly as opposed to using target or ST code like "get second button on current card...". Maybe we should try as hard as possible to make object id usage rarely needed? Then maybe the weird cases where it's necessary might simply be acceptable as long hashes?

**Comment by ApproximateIdentity** [comment]
> @ApproximateIdentity this all makes sense to me
> 
> there are three things left to do imo:
> 
>     * rebase over `master`   I merged in the [current PR](https://github.com/dkrasner/Simpletalk/pull/33) which deals with ids as well and it would be good to make sure all the tests still pass and the thing works
> 
>     * remove everything that is related to the choice to use old ids or make news one [here](https://github.com/dkrasner/Simpletalk/blob/thomas-uuid-id/js/objects/utils/serialization.js#L234)   if all ids are unique i don't see why we would ever need to make this choice moving fwd (usually done on importing a world)
> 
>     * update/re-serialize the examples

@dkrasner This all makes total sense. Will do.

**Comment by darth-cheney** [comment]
> Maybe we should try as hard as possible to make object id usage rarely needed?
  
Yeah, I think this is a good idea. The only issue is the type of references in scripts: are we referring to a specific Part, even if it moves around in its location (ie by id) or are we referring to a location that could potentially point to an arbitrary/changing part (ie, `second button of this card` when the order of buttons can change)? This is just something to keep in mind when we are writing our basic stacks

**Pull request closed without merging**